src/data/matrix/basic.lean

```diff
@@ -104,21 +119,21 @@ instance [monoid R] [add_monoid α] [distrib_mul_action R α] :
 instance [semiring R] [add_comm_monoid α] [module R α] :
   module R (matrix m n α) := pi.module _ _ _
 
-@[simp] lemma map_zero [has_zero α] [has_zero β] {f : α → β} (h : f 0 = 0) :
+@[simp] lemma map_zero [has_zero α] [has_zero β] (f : α → β) (h : f 0 = 0) :
```

Comment:
Could you explain why `f` is now explicit?

Reply:
It felt like you might want to do `rw ←map_zero f`, and then be left with convenient proof goals, rather than having to do `rw ←map_zero (_ : f 0 = 0)`. Do we have a rule that says f should be implicit?

Reply:
> It felt like you might want to do rw ←map_zero f, and then be left with convenient proof goals

OK, sounds good to me.

> Do we have a rule that says f should be implicit?

We have the guideline that parameters inferrable from explicit parameters should be implicit, but your `rw map_zero` example is convincing enough for me that the `f` is not inferrable. Choosing between explicit and implicit seems to be more of an art than a science anyway - which is why I asked. Thanks for explaining!